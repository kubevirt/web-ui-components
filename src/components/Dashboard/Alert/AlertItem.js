import React from 'react';
import PropTypes from 'prop-types';
import { Icon } from 'patternfly-react';

import { getAlertSeverity, getAlertMessage, getAlertDescription } from '../../../selectors/prometheus';

const getSeverityIcon = severity => {
  switch (severity) {
    case 'critical':
      return (
        <Icon type="fa" name="exclamation-circle" className="kubevirt-alert__icon kubevirt-alert__icon--critical" />
      );
    case 'warning':
    default:
      return (
        <Icon type="fa" name="exclamation-triangle" className="kubevirt-alert__icon kubevirt-alert__icon--warning" />
      );
  }
};

export class AlertItem extends React.PureComponent {
  render() {
    const { alert } = this.props;
    return (
      <div className="kubevirt-alert__item">
        {getSeverityIcon(getAlertSeverity(alert))}
        <div className="kubevirt-alert__item-message">{getAlertDescription(alert) || getAlertMessage(alert)}</div>
      </div>
    );
  }
}

AlertItem.propTypes = {
  alert: PropTypes.object.isRequired,
};
