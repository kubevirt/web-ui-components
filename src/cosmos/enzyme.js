import createContext from 'react-cosmos-test/enzyme';

import paths from '../../config/paths';

export default function createTestContext(args) {
  args.proxies = args.proxies || []; // workaround of react-cosmos-test bug
  return createContext({
    ...args,
    cosmosConfigPath: `${paths.config}/cosmos.config.js`,
  });
}
