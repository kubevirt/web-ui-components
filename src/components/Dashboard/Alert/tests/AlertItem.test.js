import React from 'react';
import { shallow } from 'enzyme';

import { AlertItem } from '../AlertItem';
import { default as AlertItemFixtures } from '../fixtures/AlertItem.fixture';

// eslint-disable-next-line react/prop-types
const testAlertItem = ({ props }) => <AlertItem {...props} />;

describe('<AlertItem />', () => {
  AlertItemFixtures.forEach(fixture => {
    it(`renders ${fixture.name} correctly`, () => {
      const component = shallow(testAlertItem(fixture));
      expect(component).toMatchSnapshot();
    });
  });
});
