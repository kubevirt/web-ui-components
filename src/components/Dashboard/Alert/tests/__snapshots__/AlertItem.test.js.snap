// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AlertItem /> renders Critical alert item correctly 1`] = `
<div
  className="kubevirt-alert__item"
>
  <Icon
    className="kubevirt-alert__icon kubevirt-alert__icon--critical"
    name="exclamation-circle"
    type="fa"
  />
  <div
    className="kubevirt-alert__item-message"
  >
    Critical alert
  </div>
</div>
`;

exports[`<AlertItem /> renders Unknown alert item correctly 1`] = `
<div
  className="kubevirt-alert__item"
>
  <Icon
    className="kubevirt-alert__icon kubevirt-alert__icon--warning"
    name="exclamation-triangle"
    type="fa"
  />
  <div
    className="kubevirt-alert__item-message"
  >
    unknown type alert
  </div>
</div>
`;

exports[`<AlertItem /> renders Warning alert item correctly 1`] = `
<div
  className="kubevirt-alert__item"
>
  <Icon
    className="kubevirt-alert__icon kubevirt-alert__icon--warning"
    name="exclamation-triangle"
    type="fa"
  />
  <div
    className="kubevirt-alert__item-message"
  >
    20% of the metrics targets are down.
  </div>
</div>
`;
